vm/src/builtins/set.rs

```diff
@@ -353,6 +354,45 @@ impl PySetInner {
         Ok(())
     }
 
+    fn update_internal(&self, iterable: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+        // check AnySet
+        if let Ok(any_set) = AnySet::try_from_object(vm, iterable.to_owned()) {
+            self.merge_set(any_set, vm)
+        // check Dict
+        } else if let Ok(dict) = iterable.to_owned().downcast_exact::<PyDict>(vm) {
+            self.merge_dict(dict, vm)
+        } else {
+            // add iterable that is not AnySet or Dict
+            for item in iterable.try_into_value::<ArgIterable>(vm)?.iter(vm)? {
+                self.add(item?, vm)?;
+            }
+            Ok(())
+        }
+    }
+
+    fn merge_set(&self, any_set: AnySet, vm: &VirtualMachine) -> PyResult<()> {
+        let obj = any_set.into_object();
+        let class = obj.class();
+        match extract_set(obj.as_object()) {
+            Some(other_inner) => {
+                for item in other_inner.elements() {
+                    self.add(item, vm)?;
+                }
+                Ok(())
+            }
+            None => {
+                Err(vm.new_type_error(format!("{} is not a subtype of set or frozenset", class)))
+            }
```

Comment:
this is impossible. `extract_set` always succeed. obj is always AnySet here. Try to add `as_inner()` to `AnySet`. I think `extract_set` doesn't fit to AnySet.